### ISSUE TITLE
[ACTIONS] New Writesonic, ChatSonic, Photosonic actions

### ISSUE THREAD

**Issue by RiverTeacher** [PipedreamHQ/pipedream#6054]
**Name of app / service**
Chat sonic
**Link to developer documentation**
https://docs.writesonic.com/reference/chatsonic_v2businesscontentchatsonic_post-1
**Is lack of support preventing you from building workflows, or do you have a workaround?**
Unable to build workflow due to lack of support
**Are there specific actions, or triggers, you'd like to see for this app? Please let us know here or use the Action and Trigger issue templates to open requests for each!**
Chat with Chat Sonic

Update: 4/19/2023
After discussion with the user, they want to replicate OpenAI components onto ChatSonic, however, ChatSonic is a subproduct of Writer, so certain OpenAI actions match those of Writer instead of ChatSonic, this is reflected in the following component list, which is grouped by the app actually handling a corresponding OpenAI action

**App**
ChatSonic

**Actions**
Chat

**Link to developer docs**
https://docs.writesonic.com/reference/chatsonic_v2businesscontentchatsonic_post-1

**App**
Writer

**Link to developer docs**
https://docs.writesonic.com/reference/summary_v2businesscontentsummary_post-1
https://docs.writesonic.com/reference/content-rephrase_v2businesscontentcontent-rephrase_post

**Actions**
Text Summary
Content Rephraser

**App**
Photosonic

**Link to developer docs**
https://docs.writesonic.com/reference/generate_image_v1_business_photosonic_generate_image_post-1

**Actions**
Generate Image

Context: User is requesting to build counterpart components from OpenAI in their Writer/ChatSonic/Photosonic versions.
From those Create Transction, Create Embeddings, Create Completion (Send Prompt), and Classify Items in Categories don't have an equivalent. Content Rephraser can be used for translating, as per [Tips for using Content Rephraser](https://docs.writesonic.com/docs/content-rephraser-1#tips-for-using-content-rephraser)

**Comment by pdactionddev** [comment]
Hey @RiverTeacher 

Thanks for calling that out,, however, I see that ChaSonic is properly in place: https://pipedream.com/apps/chatsonic

Can you please have a look if that works for your or other wise elaborate on what you are trying to do so that we can help you?

**Comment by RiverTeacher** [comment]
Thank you for your reply. chat sonic itself exists, but it has only 3 actions, which makes it difficult to integrate with other apps. As an example, open ai has many more actions than 3 and can be easily integrated with other apps.
I would like to see them added to chat sonic as well, if you don't mind.
Translated with DeepL https://www.deepl.com/app/?utm_source=android&utm_medium=app&utm_campaign=share-translation
![Screenshot_20230418-064622](https://user-images.githubusercontent.com/75714738/232618381-6be0a73b-b0f3-4506-9b25-cb6fee6c4d9a.png)
![Screenshot_20230418-064602](https://user-images.githubusercontent.com/75714738/232618386-25a12155-9ac2-4968-917d-624fd0a7d8f4.png)

**Comment by RiverTeacher** [comment]
Yes, we want to support the same components in Chat Sonic that are supported in OpenAI.

Triggers

Receive new emails

Actions

Chat with Chat sonic

**Comment by sergio-eliot-rodriguez** [comment]
@RiverTeacher thanks for the explanation.

Ok, so ChatSonic is part of a family of product that also includes Writesonic and Photosonic,
We have integrated Writesonic https://pipedream.com/apps/writesonic
we don't have integrated Photosonic, I created an issue for it at  https://github.com/PipedreamHQ/pipedream/issues/6098, feel free to follow that think, if you'd like.

Now, I checked the entire API documentation for Writsonic / Chatsonic / Photosonic and the following components from OpenAI are not available: 
Create Transction, Create Embeddings, Create Completion (Send Prompt), and Classify Items in Categories
You'll be able to use the Content Rephraser action component to translate, as per this Tips for using it: Create Transction, Create Embeddings, Create Completion (Send Prompt), and Classify Items in Categories

the Pipedream component development team should get to tackle these components soon, so be attentive. Thanks.

**Comment by dylburger** [comment]
Thanks for the request! Did you know that anyone can [author and publish their own Pipedream actions](https://pipedream.com/docs/components/quickstart/nodejs/actions/)? Would you be interested in helping out with that? We're happy to answer any questions or support you in any way.

If you don't have the time, would you be willing to reach out to the app provider to see if they're interested in building these? When we work with an app partner, it improves the quality of the integration, and it helps when this kind of request comes from a customer like you.

**Comment by RiverTeacher** [comment]
It's been a while. I read the documentation but did not understand it well 😭
